### ISSUE TITLE
No Man's Connect startup is stuck in 'syncing discoveries'

### ISSUE THREAD

**Issue by psouza4** [jaszhix/NoMansConnect#114]
No Man's Connect startup is stuck in 'syncing discoveries'

OPERATING SYSTEM:  Windows 10 Pro x64
SOFTWARE VERSION:  No Man's Connect v1.5.8 
SAVE DIRECTORY:  C:\Users\psouz\AppData\Roaming\HelloGames\NMS 
INSTALL DIRECTORY:  D:\SteamLibrary\steamapps\common\No Man's Sky

LOG:  https://pastebin.com/raw/AgnWs5Wp
SETTINGS.JS:  https://pastebin.com/raw/tEWFGjYF
IMAGES OF ISSUE:  https://imgur.com/a/cmFzxWv

All I did was install the app and launch it, it never got past "syncing discoveries".  I tried updating my install path, since it was not detected automatically and the program automatically restarted only to have the same issue.  According to my profile, I have 1,403 discoveries and it's been trying to sync them for an hour since the last time I interrupted.  I'm on a 128 Mbps down/12.8 Mbps up wired business-class connection that's free of saturation or throttling.  Windows Firewall is completely disabled.  Not sure what else to check.

**Comment by JANIS1289** [comment]
Have the same bug.

**Comment by jaszhix** [comment]
@psouza4 I tried to reproduce this locally with a test server using your settings file, and it loads for me. The stored locations are not accurate because it's still reading my own save file.

![image](https://user-images.githubusercontent.com/6859057/62827060-59082680-bb8d-11e9-9a8d-1b2daaf36d15.png)

Would you guys be able to [email](mailto:jaszhix@gmail.com) your NMS save file as well? I will not do any testing with it in the game or on the production server.

I can't reproduce this with my own save file, but shouldn't stall the startup process:

`Unable to update base data for location:  1360:6:-1124:0:465:1`

**Comment by Rigsly** [comment]
Same issue here, exactly as described by psouza4. Version and Save Path are basically the same. Offline mode works fine.

**Comment by Rigsly** [comment]
Ok, I just got it working by:
1. Opening NMC
2. Launching NMS / loading my save file
3. Closing NMS again.
(NMC was open in the background the whole time.)

**Comment by L-a-n-g-o-l-i-e-r-s** [comment]
jaszhix

I have the same troubles as the others, opening offline mode works but doesn’t populate the list to the right.

**Comment by jaszhix** [comment]
I released version 1.5.10, which will hopefully fix or reduce the issue. I haven't been able to reproduce it locally, and I suspect it's because the server is taking too long to process discoveries in bulk with the current user load. This mainly affects new users with existing save files, as they have the most to sync. So if anyone was running into this in 1.5.9, please let me know if 1.5.10 works or not.

**Comment by psouza4** [comment]
I forgot to check back and say I got the issue resolved the same way@Rigsly or @L-a-n-g-o-l-i-e-r-s did -- closing No Man's Sky.  No Man's Connect does not want to touch the save game data if NMS is running, which is unfortunate.

**Comment by bassj18** [comment]
got same error 
![Untitled](https://user-images.githubusercontent.com/54414889/63538652-37edf100-c510-11e9-8c76-b5aea8988e50.jpg)
1566497390066:    State initialized is in log file !

**Comment by jaszhix** [comment]
Ok, this won't help me reproduce this, or fix it. I need someone's save file.

**Comment by jaszhix** [comment]
> No Man's Connect does not want to touch the save game data if NMS is running, which is unfortunate.

That's a Windows limitation, not a No Man's Connect limitation. Windows will not let a program write to a file if it's locked, that error is just handled and displayed in the log to prevent the polling from breaking.

**Comment by psouza4** [comment]
> > No Man's Connect does not want to touch the save game data if NMS is running, which is unfortunate.
> 
> That's a Windows limitation, not a No Man's Connect limitation. Windows will not let a program write to a file if it's locked, that error is just handled and displayed in the log to prevent the polling from breaking.

1. That's (a) only if the file is locked and (b) only if the application is trying to write to the locked file.  Is No Man's Connect trying to write to my game save just by starting it up?  I am not trying to make changes with NMC, just trying to start the program.
2. Nowhere in the log does it seem to indicate that there's a file lock that NMC is waiting for.
3. Nowhere in the interface does it indicate that there's a file lock that NMC is waiting for.  It sits there forever informing the user it's trying to do something.

This begs to question the usefulness of a companion mapping program if it can't be used while playing the game.  Unless you misunderstand when this error occurs, I have serious doubts that NMC is trying to write to a save game on its startup.  However, if this _is_ actually the case and NMC wants to write to your save game on startup, I'd request two things: (a) that users be made aware that their save game files will be altered immediately upon using the program without expressly asking to save anything and (b) that you simply write some logic that copies the locked file to a temporary/shadow version so that read-only functionality can be enabled (like mapping features) while the game is being played.

**Comment by jaszhix** [comment]
NodeJS is throwing EBUSY, this mean the file is _locked_ by the game. I did not see this error in your logs, however. NMC is not writing to the save file unless you teleport, restore a base, or use the hidden cheat menu.

> This begs to question the usefulness of a companion mapping program if it can't be used while playing the game.

This begs the question the usefulness of this bug report if there's not enough information here for me to reproduce it.

**Comment by psouza4** [comment]
> NodeJS is throwing EBUSY, this mean the file is _locked_ by the game. I did not see this error in your logs, however. NMC is not writing to the save file unless you teleport, restore a base, or use the hidden cheat menu.
> 
> > This begs to question the usefulness of a companion mapping program if it can't be used while playing the game.
> 
> This begs the question the usefulness of this bug report if there's not enough information here for me to reproduce it.

Please see my original post: I've given you all the information about steps I took.  At no point did I get to operate the application (such as attempting to "teleport", "restore a base", or "use the hidden cheat menu") because the application refused to finish loading my save file.

I produced exactly the requested log, settings, and other pertinent information you request in the bug report template.  If there's additional information you want from me, by all means let me know -- seems pretty reproducible though and seems an easy enough fix.  Instead of opening files RW, just open them as R until any operation that requires writing to the file promotes a RW open.

**Comment by jaszhix** [comment]
> I produced exactly the requested log, settings, and other pertinent information you request in the bug report template
> Would you guys be able to email your NMS save file as well? I will not do any testing with it in the game or on the production server.

I'll leave this open for a few more days, but I don't leave issues open for my requests to be ignored while "me too!" comments pile up.

**Comment by jaszhix** [comment]
To clarify, I'm not dismissing your point about the file being locked, and opening it read-only seems like a good idea. It however, is not supported by the information you provided, and should be opened as a separate issue. The request timing out seems to be the case here because it only affects new users.

**Comment by psouza4** [comment]
> Your logs don't support your theory of this being a lock issue. If that is the case, upload new logs.

I agree, it doesn't, [but you said it did](https://github.com/jaszhix/NoMansConnect/issues/114#issuecomment-524025012).

That was your theory of this being a lock issue, based on my summary of the issue where No Man's Connect will not load a save file while No Man's Sky is running a Windows limitation because the file is locked and Windows won't allow writing to it.  I don't think this is actually what's happening, and I agree that the logs don't support this, but it's certainly not my theory.

> > > I produced exactly the requested log, settings, and other pertinent information you request in the bug report template
> > Would you guys be able to email your NMS save file as well? I will not do any testing with it in the game or on the production server.
> 
> I'll leave this open for a few more days, but I don't leave issues open for my requests to be ignored while "me too!" comments pile up.
I am really confused here.  Honestly.

I'm a little uncomfortable mailing you my save data -- I don't know what kind of personally-identifying information it contains, but since you made the blanket promise not to load up the game using our saves, that signals me that we wouldn't want you to for some reason.  So there's skepticism about doing so.

Although.... that's likely moot anyway as my saves have changed hundreds of times in the last week.  It may or may not even have this issue anymore as I haven't tried opening NMC since.  However, if it does, you would most probably be _required_ to load into the game using my save file since the problem is only apparent when the NMS game is also running.

However, the parameters of the bug report *have* changed.  As I reported, and have others, it seems to be an issue with having the NMS game running.  Have you tried reproducing this issue with the NMS game running?  If you have and NMC loads the same save file fine without hanging, then you likely _will_ need my save files (if they still exhibit the issue) and you will definitely have to load the game in NMS to test with.  That's something I can do.

I appreciate that you're taking your time to support freeware and understand that I'm also taking the time to be as detailed with my report as I can be for you.

**Comment by jaszhix** [comment]
> I agree, it doesn't, but you said it did.

No, I'm just referring to the error you're referencing (but didn't update your post with logs for), because I wrote that error handling in the code a long time ago. 

>  I don't know what kind of personally-identifying information it contain

At most Steam usernames, and those are not personally identifiable. I only request it being emailed as a courtesy to you, as other people might be able to exploit game bugs and mess up your accounting with the NMS servers.

> Have you tried reproducing this issue with the NMS game running?

Yes, I start up NMC while NMS is running all the time. I wouldn't know where to start reproducing this, and the goal post can't keep moving if we've identified that the bug is server-related - as it appears to stop occurring once the profile is complete with discoveries.

**Comment by psouza4** [comment]
My oldest backup does not exhibit the issue from an account that NMC has already accessed.

I tried deleting all of the NMC cache and settings out of AppData, loading up the NMS game with my old save, and starting NMC but it doesn't show the issue.  I can't reset myself back to a fresh account that's never used NMC, unfortunately.

You may need a save file from someone who's never used NMC and try loading it using just the default NMC settings while NMS is already running that save game.  That's really all I can suggest because now that I'm past the error, I can't reproduce it (even though it plagued me for hours and hours across multiple NMC sessions until I had finally finished playing that session of NMS).

Alternatively, is there a NMC server that you're synchronizing data with?  If so, we could reset that, too, and try.

**Comment by jaszhix** [comment]
I have a separate dev instance of the server running with a backup of the database. Just recently, I have had to delete my own profile on it locally to reproduce #118. I could forward the ports, but the limited bandwidth wouldn't reproduce a production environment. It sounds like it's combination of the data needing to be new to the server, with a save file that triggers the bug. If nobody wants to cooperate with save files, then this bug is not a guaranteed fix.

**Comment by bassj18** [comment]
[st_76561197965437052.zip](https://github.com/jaszhix/NoMansConnect/files/3531637/st_76561197965437052.zip)
her my save file is that helps ! i did start my game join co op friend not sure if that got do something with it !

**Comment by MelStol** [comment]
[st_76561198085937843.zip](https://github.com/jaszhix/NoMansConnect/files/3532614/st_76561198085937843.zip)
Here is my save file as well. Having similar issues. Offline mode loads the map, not anything else (is this normal for offline mode?).

**Comment by jaszhix** [comment]
Thanks for the save files. I'm able to reproduce the startup routine getting stuck now, and looking into it. 

> Offline mode loads the map, not anything else (is this normal for offline mode?).

If it says "Explorer" as your username, then it's not fully set up to work correctly. Otherwise, it should work and cache your own locations as you play the game, but you won't see anybody else's locations because there hasn't been a successful download of them while online.

**Comment by MelStol** [comment]
My username is loaded properly when used in offline mode.
![Screenshot (101a)](https://user-images.githubusercontent.com/28700458/63564409-cd41b300-c519-11e9-890e-b1037bf1e094.png)

**Comment by jaszhix** [comment]
Pushed a release, this fixed it for me locally - https://github.com/jaszhix/NoMansConnect/releases/download/v1.8.2/NoMansConnect.Setup.1.8.2.exe

Please let me know if there's still issues related to this. For offline mode, that should be a new issue. Thanks.

**Comment by MelStol** [comment]
1.8.2 has fixed this. It has also fixed offline mode for me as well, so that's neat.

**Comment by bassj18** [comment]
> 
> 
> Pushed a release, this fixed it for me locally - https://github.com/jaszhix/NoMansConnect/releases/download/v1.8.2/NoMansConnect.Setup.1.8.2.exe
> 
> Please let me know if there's still issues related to this. For offline mode, that should be a new issue. Thanks.

hi mate let know my all fix and working good !
